src/Microsoft.NetCore.Analyzers/Core/Security/ApprovedCipherModeAnalyzer.cs

```diff
@@ -0,0 +1,98 @@
+        private static readonly LocalizableString Message = new LocalizableResourceString(
+            nameof(SystemSecurityCryptographyResources.ApprovedCipherModeMessage),
+            SystemSecurityCryptographyResources.ResourceManager,
+            typeof(SystemSecurityCryptographyResources));
+        private static readonly LocalizableString Description = new LocalizableResourceString(
+            nameof(SystemSecurityCryptographyResources.ApprovedCipherModeDescription),
+            SystemSecurityCryptographyResources.ResourceManager,
+            typeof(SystemSecurityCryptographyResources));
+
+        internal static DiagnosticDescriptor Rule =
+            CreateDiagnosticDescriptor(DiagnosticId, Title, Message, Description);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        internal string CipherModeTypeMetadataName =
+            "System.Security.Cryptography.CipherMode";
+
+        internal ImmutableHashSet<string> UnsafeCipherModes = ImmutableHashSet.Create(
+                StringComparer.Ordinal,
+                "ECB",
+                "OFB",
+                "CFB");
+
+        private static DiagnosticDescriptor CreateDiagnosticDescriptor(string ruleId, LocalizableString title, LocalizableString message, LocalizableString description, string uri = null)
+        {
+            return new DiagnosticDescriptor(
+                ruleId,
+                title,
+                message,
+                DiagnosticCategory.Security,
+                DiagnosticHelpers.DefaultDiagnosticSeverity,
+                isEnabledByDefault: DiagnosticHelpers.EnabledByDefaultIfNotBuildingVSIX,
+                description: description,
+                helpLinkUri: uri,
+                customTags: WellKnownDiagnosticTags.Telemetry);
+        }
+
+        public sealed override void Initialize(AnalysisContext context)
+        {
+            Debug.Assert(CipherModeTypeMetadataName != null);
+            Debug.Assert(UnsafeCipherModes != null);
+            Debug.Assert(!UnsafeCipherModes.IsEmpty);
+
+            context.EnableConcurrentExecution();
+
+            // Security analyzer - analyze and report diagnostics on generated code.
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
+
+            context.RegisterCompilationStartAction(
+                (CompilationStartAnalysisContext compilationStartAnalysisContext) =>
+                {
+                    INamedTypeSymbol cipherModeTypeSymbol =
+                        compilationStartAnalysisContext.Compilation.GetTypeByMetadataName(CipherModeTypeMetadataName);
+                    if (cipherModeTypeSymbol == null)
+                    {
+                        return;
+                    }
+                    compilationStartAnalysisContext.RegisterOperationAction(
```

Comment:
Nit: most files have the style of a blank line after closing brace (unless it's another closing brace)